File: app/src/main/java/com/example/myappcontacts/data/repositories/contacts/ContactsRepository.java
package com.example.myappcontacts.data.repositories.contacts;

import com.example.myappcontacts.data.dao.contacts.db.ContactsModel;
import com.example.myappcontacts.data.providers.database.IDataBaseProvider;

import java.util.UUID;

import io.reactivex.Completable;
import io.reactivex.Single;

public class ContactsRepository implements IContactsRepository {

    private IDataBaseProvider mDataBaseProvider;

    public ContactsRepository(IDataBaseProvider sqLiteDbProvider){
        mDataBaseProvider = sqLiteDbProvider;
    }

    @Override
    public Single<ContactsModel> loadContact(UUID contactId) {
        return mDataBaseProvider.loadContact(contactId);
    }

    @Override
    public Completable updateContact(ContactsModel contactsModel) {
        return mDataBaseProvider.updateContact(contactsModel);
    }
}

File: app/src/main/java/com/example/myappcontacts/data/repositories/contacts/ContactsRepository.kt
package com.example.myappcontacts.data.repositories.contacts

import com.example.myappcontacts.data.dao.contacts.db.ContactsModel
import com.example.myappcontacts.data.providers.database.IDataBaseProvider
import io.reactivex.Completable
import io.reactivex.Single
import java.util.*

class ContactsRepository(private val dataBaseProvider: IDataBaseProvider) : IContactsRepository {
    override fun loadContact(contactId: UUID): Single<ContactsModel> {
        return dataBaseProvider.loadContact(contactId)
    }

    override fun updateContact(contactsModel: ContactsModel): Completable {
        return dataBaseProvider.updateContact(contactsModel)
    }
}